### ISSUE TITLE
Silenced error detection not working correctly

### ISSUE THREAD

**Issue by alexander-schranz** [getsentry/sentry-php#1050]
We have basically default installation of sentry in our symfony application, currently the following sentry packages installed

```bash
sentry/sdk: 2.1.0
sentry/sentry: 2.4.0 
sentry/sentry-symfony:3.5.1
```

Sentry is running on its default configuration we just have the dsn configured:

```yaml
sentry:
      dsn: "%env(SENTRY_DSN)%"
```

But we still get the following errors the following sentry errors:

> Sentry\Exception\FatalErrorException
> Compile Warning: Unterminated comment starting line 41
> /vendor/phpcr/phpcr-migrations/lib/MigratorUtil.php at line 41

```php
            // Read entire lines to prevent keyword truncation
            for ($line = 0; $line <= 20; $line++) {
                $buffer .= fgets($fp);
            }
            $tokens = @token_get_all($buffer); // this is line 41 which fails
```

<details>

<summary>Screenshot</summary>

<img width="812" alt="Bildschirmfoto 2020-07-16 um 15 57 39" src="https://user-images.githubusercontent.com/1698337/87679979-268f8200-c77d-11ea-99e6-8d3b4ade0354.png">

</details>

as we don't have `capture_silenced_errors` activated this error should not appear.

What is strange that it is detected as `FatalErrorException` and if I look at the current code [here](https://github.com/getsentry/sentry-php/blob/6798aca3375fcad16b763614c8cba7e11f06b0d0/src/Integration/ErrorListenerIntegration.php#L73), it should be a `SilencedErrorException` so it is not logged, so maybe there is the error but I'm not sure.

The code throwing this warning is basically very simple: `token_get_all("<?php /*");`.

**Comment by Jean85** [comment]
I think this is a bug on the underlying `sentry-php` lib. I'll move the issue accordingly.

**Comment by Jean85** [comment]
Confirmed, reproduced in #1051.

**Comment by ste93cry** [comment]
>What is strange that it is detected as `FatalErrorException` and if I look at the current code here, it should be a `SilencedErrorException` so it is not logged, so maybe there is the error but I'm not sure.

Technically speaking you are absolutely right: the error should be silenced. However, practically, this is a kind of fatal error and PHP does not handle it through the error handler, so the `capture_silenced_errors` option gets ignored as it's used only by the `ErrorListenerIntegration` class. Even if we were to use that option in the `FatalErrorListenerIntegration` class we would not be able to distinguish whether it was suppressed because PHP doesn't allow suppression of fatal errors. I don't really have an idea of how we can fully solve this issue, as ignoring `E_CORE_WARNING` and `E_COMPILE_WARNING` errors would then cause those to never be catched, while adjusting the code to log them as errors instead of fatals still won't ignore them if they were suppressed.

**Comment by Jean85** [comment]
Closing this, since the reproducer in #1051 is basically the same as this one: https://github.com/getsentry/sentry-php/blob/8fea3e3ec6b27bfef7899efa9ecf26b1396a1b31/tests/phpt/error_handler_respects_capture_silenced_errors_option.phpt#L56-L64

So fixed in #1183
